core/benches/ops/async.rs

```diff
@@ -0,0 +1,222 @@
+pub async fn op_async_yield_deferred_nofast() {
+  tokio::task::yield_now().await
+}
+
+#[op2(async(lazy), fast)]
+pub async fn op_async_void_lazy() {}
+
+#[op2(async(lazy), nofast)]
+pub async fn op_async_void_lazy_nofast() {}
+
+#[op2(async(deferred), fast)]
+pub async fn op_async_void_deferred() {}
+
+#[op2(async(deferred), nofast)]
+pub async fn op_async_void_deferred_nofast() {}
+
+fn bench_op(
+  b: &mut Bencher,
+  count: usize,
+  op: &str,
+  arg_count: usize,
+  call: &str,
+) {
+  #[cfg(not(feature = "unsafe_runtime_options"))]
+  unreachable!(
+    "This benchmark must be run with --features=unsafe_runtime_options"
+  );
+
+  let tokio = tokio::runtime::Builder::new_current_thread()
+    .build()
+    .unwrap();
+  let mut runtime = JsRuntime::new(RuntimeOptions {
+    extensions: vec![testing::init_ops_and_esm()],
+    // We need to feature gate this here to prevent IDE errors
+    #[cfg(feature = "unsafe_runtime_options")]
+    unsafe_expose_natives_and_gc: true,
+    ..Default::default()
+  });
+  let err_mapper =
+    |err| generic_error(format!("{op} test failed ({call}): {err:?}"));
+
+  let args = (0..arg_count)
+    .map(|n| format!("arg{n}"))
+    .collect::<Vec<_>>()
+    .join(", ");
+
+  // Prime the optimizer
+  runtime
+    .execute_script(
+      "",
+      FastString::Owned(
+        format!(
+          r"
+async function run() {{
+  const LARGE_STRING_1000000 = '*'.repeat(1000000);
+  const LARGE_STRING_1000 = '*'.repeat(1000);
+  const LARGE_STRING_UTF8_1000000 = '\u1000'.repeat(1000000);
+  const LARGE_STRING_UTF8_1000 = '\u1000'.repeat(1000);
+  const BUFFER = new Uint8Array(1024);
+  const ARRAYBUFFER = new ArrayBuffer(1024);
+  const {{ {op}: op }} = Deno.core.ensureFastOps();
+  const {{ op_make_external }} = Deno.core.ensureFastOps();
+  const EXTERNAL = op_make_external();
+  function {op}({args}) {{
+    op({args});
+  }}
+  let accum = 0;
+  let __index__ = 0;
+  %PrepareFunctionForOptimization({op});
+  {call};
+  %OptimizeFunctionOnNextCall({op});
+  {call};
+
+  async function bench() {{
+    let accum = 0;
+    for (let __index__ = 0; __index__ < {count}; __index__++) {{ {call} }}
+    return accum;
+  }}
+  
+  %PrepareFunctionForOptimization(bench);
+  await bench();
+  %OptimizeFunctionOnNextCall(bench);
+  await bench();
+
+  return bench;
+}}
+        "
+        )
+        .into(),
+      ),
+    )
+    .map_err(err_mapper)
+    .unwrap();
+  let guard = tokio.enter();
+  let run = runtime.execute_script("", ascii_str!("run()")).unwrap();
+  let bench = tokio.block_on(runtime.resolve_value(run)).unwrap();
+  let mut scope = runtime.handle_scope();
+  let bench: v8::Local<v8::Function> =
+    v8::Local::new(&mut scope, bench).try_into().unwrap();
+  let bench = v8::Global::new(&mut scope, bench);
+  drop(scope);
+  drop(guard);
+  b.iter(move || {
+    tokio.block_on(async {
+      let guard = tokio.enter();
+      runtime.call_and_await(&bench).await.unwrap();
+      drop(guard);
+    });
+  });
```

Comment:
Not a blocker, but have you profiled it using `samply`? One time I did that for other benchmark I saw a surprising profile where most of the time is spent in runtime creation/dropping.

Reply:
I haven't in this case, but I included the baseline so we can see the overhead of running the same overhead code but with no work. It really comes down to spinning the event loop being slow :/

```
test baseline                             ... bench:       1,194 ns/iter (+/- 34)
test bench_op_async_void                  ... bench:     149,005 ns/iter (+/- 3,891)
```